lib/Conversion/GPUToROCM/GenerateHSACOAccessors.cpp

```diff
@@ -0,0 +1,135 @@
+/// functions for them. The external kernel functions (also generated by the
+/// outlining pass) are annotated with the symbol of the hsaco accessor.
+class GpuGenerateHSACOAccessorsPass
+    : public ModulePass<GpuGenerateHSACOAccessorsPass> {
+private:
+  // Inserts a global constant string containing `blob` into the parent module
+  // of `kernelFunc` and generates the function that returns the address of the
+  // first character of this string.
+  void generate(FuncOp kernelFunc, StringAttr hsacoBlob) {
+    auto stubFunc = getModule().lookupSymbol<FuncOp>(kernelFunc.getName());
+    if (!stubFunc) {
+      kernelFunc.emitError(
+          "corresponding external function not found in parent module");
+      return signalPassFailure();
+    }
+
+    Location loc = stubFunc.getLoc();
+    SmallString<128> nameBuffer(stubFunc.getName());
+    auto module = stubFunc.getParentOfType<ModuleOp>();
+    assert(module && "function must belong to a module");
+
+    // Insert the getter function just after the original function.
+    OpBuilder moduleBuilder(module.getBody(), module.getBody()->begin());
+    moduleBuilder.setInsertionPointAfter(stubFunc.getOperation());
+    auto getterType = moduleBuilder.getFunctionType(
+        llvm::None, LLVM::LLVMType::getInt8PtrTy(llvmDialect));
+    nameBuffer.append(rocm::kHSACOGetterSuffix);
+    auto result = moduleBuilder.create<FuncOp>(
+        loc, StringRef(nameBuffer), getterType, ArrayRef<NamedAttribute>());
+    Block *entryBlock = result.addEntryBlock();
+
+    // Drop the getter suffix before appending the storage suffix.
+    nameBuffer.resize(stubFunc.getName().size());
+    nameBuffer.append(rocm::kHSACOStorageSuffix);
+
+    // Obtain the address of the first character of the global string containing
+    // the cubin and return from the getter.
+    OpBuilder builder(entryBlock);
+    Value *startPtr = LLVM::createGlobalString(
+        loc, builder, StringRef(nameBuffer), hsacoBlob.getValue(), llvmDialect);
+    builder.create<LLVM::ReturnOp>(loc, startPtr);
+
+    // Store the name of the getter on the function for easier lookup.
+    stubFunc.setAttr(rocm::kHSACOGetterAnnotation,
+                     builder.getSymbolRefAttr(result));
+  }
+
+public:
+  // Run the dialect converter on the module.
+  void runOnModule() override {
+    llvmDialect = getContext().getRegisteredDialect<LLVM::LLVMDialect>();
+
+    auto modules = getModule().getOps<ModuleOp>();
+    for (auto module : llvm::make_early_inc_range(modules)) {
+      if (!module.getAttrOfType<UnitAttr>(
+              gpu::GPUDialect::getKernelModuleAttrName()))
+        continue;
+      for (auto func : module.getOps<FuncOp>()) {
+        if (StringAttr hsacoBlob =
+                func.getAttrOfType<StringAttr>(rocm::kHSACOAnnotation))
+          generate(func, hsacoBlob);
+      }
+      module.erase();
+    }
+
+    // dump the MLIR
+    if (getenv("MLIR_ROCM_DUMP_AFTER_GEN_HSACO_GETTER")) {
```

Comment:
Same comment as the other file.

Reply:
this was meant for personal debug use only and not meant to make it into the upstream commit. 
will remove it (and other such instances) the next push